src/tests/testHttpRange.cc

```diff
@@ -126,26 +151,13 @@ testRangeCanonization()
 }
 
 int
-main(int, char **)
+main(int argc, char **argv)
 {
-    try {
-        Mem::Init();
-        /* enable for debugging to console */
-        // Debug::debugOptions = xstrdup("ALL,1 64,9");
-        // Debug::BanCacheLogUse();
-        testRangeParser("bytes=0-3");
-        testRangeParser("bytes=-3");
-        testRangeParser("bytes=1-");
-        testRangeParser("bytes=0-3, 1-, -2");
-        testRangeIter();
-        testRangeCanonization();
-    } catch (const std::exception &e) {
-        printf("Error: dying from an unhandled exception: %s\n", e.what());
-        return EXIT_FAILURE;
-    } catch (...) {
-        printf("Error: dying from an unhandled exception.\n");
-        return EXIT_FAILURE;
-    }
```

Comment:
I can't be sure about the actual intent of code that isn't mine,
but this can look similar to a one-off left in while doing some
sort of TDD, to facilitate development.

Anyway, here's the tests you require, built off master.
```patch
@@ -32,6 +34,8 @@ testRangeParser(char const *rangestring)
     assert (*pos);

     delete range;
+
+    throw std::range_error("injected exception");
 }

 static HttpHdrRange *
```

results in 
```
SKIP: mem/libmem.la MemPools (not implemented).
SKIP: security/libsecurity.la PeerOptions (not implemented).
SKIP: mem/libmem.la Init (not implemented).
Error: dying from an unhandled exception: injected exception
```

.
```patch
+struct SomeRandomException {};
+
 static void
 testRangeParser(char const *rangestring)
 {
@@ -32,6 +36,8 @@ testRangeParser(char const *rangestring)
     assert (*pos);

     delete range;
+
+    throw SomeRandomException();
 }

 static HttpHdrRange *
```
results in:
```
SKIP: mem/libmem.la MemPools (not implemented).
SKIP: security/libsecurity.la PeerOptions (not implemented).
SKIP: mem/libmem.la Init (not implemented).
Error: dying from an unhandled exception.
```

With this PR applied, the same code changes result in:
```
SKIP: mem/libmem.la MemPools (not implemented).
SKIP: security/libsecurity.la PeerOptions (not implemented).
SKIP: mem/libmem.la Init (not implemented).
.E..
##Failure Location unknown## : Error
Test name: TestHttpRange::testRangeParser
uncaught exception of type std::exception (or derived).
- injected exception

Failures !!!
Run: 3   Failure total: 1   Failures: 0   Errors: 1
```

and
```
SKIP: mem/libmem.la MemPools (not implemented).
SKIP: security/libsecurity.la PeerOptions (not implemented).
SKIP: mem/libmem.la Init (not implemented).
.E..
##Failure Location unknown## : Error
Test name: TestHttpRange::testRangeParser
uncaught exception of unknown type

Failures !!!
Run: 3   Failure total: 1   Failures: 0   Errors: 1
```

respectively.
Seems superior to me in every regard, do you agree?

Reply:
> Seems superior to me in every regard, do you agree?

Suffice to say that the output from PR code is sufficiently good to proceed with removal of those custom `catch` clauses.

To answer this change request question, one would have to test official code with those  custom `catch` clauses _removed_, but I failed to spell out that part in the change request, and we probably do not care enough to spend more time to find out the answer to that question anyway... I am resolving this change request.